### PR TITLE
[Fix] #916 expection string type classes

### PR DESCRIPTION
**Pull request by ShoupingShan**
expection occurs when classes variable is a string file path

Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

fix a bug in custom.py, Exception occurs when `classes` is a string type file path .

## Modification

modify `classes` to `class_names` in https://github.com/ShoupingShan/mmsegmentation/blob/1df3a93d0d169c4bd7bc538c378ed9f2d43d50d6/mmseg/datasets/custom.py#L322 and https://github.com/ShoupingShan/mmsegmentation/blob/1df3a93d0d169c4bd7bc538c378ed9f2d43d50d6/mmseg/datasets/custom.py#L333

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

No

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by Junjun2016** [comment]
Hi @ShoupingShan 
Thanks for your contribution.
Could you please add some unittests [here](https://github.com/open-mmlab/mmsegmentation/blob/master/tests/test_data/test_dataset.py).
If the code coverage and unittests are sufficient, we could avoid many bugs and BC-Breaking.

**Comment by Junjun2016** [comment]
Hi @ShoupingShan 
The CI failed.
Please fix the lint error by using pre-commit hook and unittests error.

**Comment by Junjun2016** [comment]
The CI is still failed.
Could you please fix it?

**Comment by Junjun2016** [comment]
Please refer to https://github.com/open-mmlab/mmsegmentation/blob/master/.github/CONTRIBUTING.md.
```shell
pre-commit run --all-files
```

**Comment by Junjun2016** [comment]
Hi @ShoupingShan 
It is better to create a PR from a new branch instead of the master branch.

File: tests/test_data/test_dataset.py [hunk 1]
```diff
@@ -26,6 +27,40 @@ def test_classes():
         get_classes('unsupported')
 
 
+def test_classes_file_path():
+    train_pipeline = [dict(type='LoadImageFromFile')]
+    classes_path = '../data/classes.txt'
```
Comment (Junjun2016) [inline tests/test_data/test_dataset.py:32]: Please use `tempfile.NamedTemporaryFile()`.
You can refer to 
https://github.com/open-mmlab/mmsegmentation/blob/186a1fc88e31f294adbabb677608240aa2221d2a/tests/test_models/test_losses/test_dice_loss.py#L28

File: tests/test_data/test_dataset.py [hunk 2]
```diff
@@ -26,6 +27,40 @@ def test_classes():
...
+    # classes.txt with full categories
+    categories = get_classes('cityscapes')
+    with open(classes_path, 'w') as f:
+        f.write('\n'.join(categories))
+
+    kwargs = dict(
+        pipeline=train_pipeline,
+        img_dir='./',
+        classes=classes_path
+    )
```
Comment (Junjun2016) [inline tests/test_data/test_dataset.py:43]: image

use yapf to format the code (pre-commit hook)

**Comment by Junjun2016** [comment]
Please also rename the PR title to make it more accurate and understandable.

**Comment by Junjun2016** [comment]
Please also rename the PR title to make it more accurate and understandable.

**Comment by Junjun2016** [comment]
Please also rename the PR title to make it more accurate and understandable.

File: tests/test_data/test_dataset.py (hunk 2 above)
Comment (Junjun2016) [reply tests/test_data/test_dataset.py:43]: https://github.com/open-mmlab/mmsegmentation/pull/918/checks?check_run_id=3735897729

File: tests/test_data/test_dataset.py (hunk 1 above)
Comment (ShoupingShan) [reply tests/test_data/test_dataset.py:32]: ok, I will modify the test sample.

Comment (Junjun2016) [reply tests/test_data/test_dataset.py:32]: https://github.com/open-mmlab/mmcv/blob/c47c9196d067a0900b7b8987a8e82768edab2fff/tests/test_fileio.py

**Comment by ShoupingShan** [comment]
I will open a new pull request from a new branch

**Pull request closed without merging**